tree/ntuple/v7/inc/ROOT/RNTupleFillContext.hxx

```diff
@@ -81,18 +82,35 @@ private:
 public:
    ~RNTupleFillContext();
 
-   /// Fill an entry into this context.  This method will perform a light check whether the entry comes from the
-   /// context's own model.
+   /// Fill an entry into this context, but don't commit the cluster. The calling code must pass an RNTupleFillStatus
+   /// and check RNTupleFillStatus::ShouldCommitCluster.
+   ///
+   /// This method will perform a light check whether the entry comes from the context's own model.
+   ///
    /// \return The number of uncompressed bytes written.
-   std::size_t Fill(REntry &entry)
+   std::size_t FillNoCommit(REntry &entry, RNTupleFillStatus &status)
```

Comment:
Should we add the return value to the `RNTupleFillStatus` object and return`void`?

Reply:
Not sure. I think at some point I had the idea that we could also provide a `RNTupleFillStatus GetFillStatus() const` without actually filling anything and then I wouldn't know what to put into that field. But we can also reconsider.

Reply:
Changed as discussed offline: `RNTupleFillStatus` now has a field `fLastEntrySize` and accessor `GetLastEntrySize`. Even if we want to provide a `GetFillStatus()` API later on, this will have clear semantics and can be easily remembered in the `RNTupleFillContext.